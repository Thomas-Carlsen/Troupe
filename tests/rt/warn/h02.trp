(* 2019-01-03: testing errors in handlers. That the guard throws  *)
(* an exception but it does not terminate the thread. *)

let fun foo (x) = x + 1 > 0 
    val pid =
        spawn ( fn () => 
                    receive [ hn (x,sender) when foo x => print x ]
              )
              
    val _ = send (pid, (  "" , self()) )
in 0    
end 
