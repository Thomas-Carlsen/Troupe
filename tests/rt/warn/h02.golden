2019-03-07T12:08:28.462Z [RTM] [32minfo[39m: Skipping network creation. Observe that all external IO operations will yield a runtime error.
>>> Main thread finished with value: 0@{}%{}
[33mWarning: runtime exception in the handler or sandbox: value 1@{}%{} is not a string[39m
""
